lib/devices_api/users/commands/create.ex

```diff
@@ -2,36 +2,29 @@ defmodule DevicesApi.Users.Commands.Create do
   @moduledoc """
   Creates a new user
   """
-  alias DevicesApi.Users.Schemas.User
   alias DevicesAPI.Repo
+  alias DevicesApi.Users.Inputs.SignupRequestInput
+  alias DevicesApi.Users.Schemas.User
 
   @type user_params :: %{
           name: String.t(),
-          passord: String.t(),
+          password: String.t(),
           email: String.t()
         }
 
-  @doc """
-  Inserts an user into the database.
-
-  ## Examples
-
-
-    iex> user_params = %{name: "beatriz", password: "123457", email: "beatriz@teste.com.br"}
-
-    iex> {:ok, %User{}} = execute(user_params)
-
-    iex> {:error, %Ecto.Changeset{}} = Users.Create.execute(%{})
-
-  """
-  @spec execute(user_params()) :: {:error, Ecto.Changeset.t()} | {:ok, Ecto.Schema.t()}
-  def execute(params) when is_map(params) do
-    with %Ecto.Changeset{valid?:  true} = changeset <- User.changeset(params),
+  @spec execute(input :: SignupRequestInput.t()) ::
+          {:error, Ecto.Changeset.t()} | {:ok, Ecto.Schema.t()}
+  def execute(%SignupRequestInput{} = input) do
+    with %Ecto.Changeset{valid?: true} = changeset <- do_changeset(input),
          {:ok, user} <- Repo.insert(changeset) do
       {:ok, user}
     else
-      %Ecto.Changeset{valid?:  false} = changeset -> {:error, changeset}
+      %Ecto.Changeset{valid?: false} = changeset -> {:error, changeset}
       {:error, _} = err -> err
     end
   end
+
+  defp do_changeset(%{name: name, email: email, password: password}) do
+    User.changeset(%{name: name, email: email, password: password})
```

Comment:
QUal é a vantagem de ter uma função a parte fazendo isso? O que essa função está resolvendo?

```suggestion
  defp do_changeset(%{name: name, email: email, password: password} = params) do
    User.changeset(params)
```